formwidgets/MLNestedForm.php

```diff
@@ -0,0 +1,195 @@
+<?php namespace Winter\Translate\FormWidgets;
+
+use Backend\FormWidgets\NestedForm;
+use Winter\Translate\Models\Locale;
+use Winter\Storm\Html\Helper as HtmlHelper;
+use ApplicationException;
+use Request;
+
+/**
+ * ML NestedForm
+ * Renders a multi-lingual nestedform field.
+ *
+ * @package rainlab\translate
+ * @author Luke Towers
+ */
+class MLNestedForm extends NestedForm
+{
+    use \Winter\Translate\Traits\MLControl;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected $defaultAlias = 'mlnestedform';
+
+    /**
+     * {@inheritDoc}
+     */
+    public function init()
+    {
+        parent::init();
+        $this->initLocale();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function render()
+    {
+        $this->actAsParent();
+        $parentContent = parent::render();
+        $this->actAsParent(false);
+
+        if (!$this->isAvailable) {
+            return $parentContent;
+        }
+
+        $this->vars['nestedform'] = $parentContent;
+        return $this->makePartial('mlnestedform');
+    }
+
+    public function prepareVars()
+    {
+        parent::prepareVars();
+        $this->prepareLocaleVars();
+    }
+
+    /**
+     * Returns an array of translated values for this field
+     * @return array
+     */
+    public function getSaveValue($value)
+    {
+        $this->rewritePostValues();
+        return $this->getLocaleSaveValue($value);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function loadAssets()
+    {
+        $this->actAsParent();
+        parent::loadAssets();
+        $this->actAsParent(false);
+
+        if (Locale::isAvailable()) {
+            $this->loadLocaleAssets();
+            $this->addJs('js/mlnestedform.js');
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getParentViewPath()
+    {
+        return base_path().'/modules/backend/formwidgets/nestedform/partials';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getParentAssetPath()
+    {
+        return '/modules/backend/formwidgets/nestedform/assets';
+    }
+
+    public function onSwitchItemLocale()
+    {
+        if (!$locale = post('_nestedform_locale')) {
+            throw new ApplicationException('Unable to find a nestedform locale for: '.$locale);
+        }
+
+        // Store previous value
+        $previousLocale = post('_nestedform_previous_locale');
+        $previousValue = $this->getPrimarySaveDataAsArray();
+
+        // Update widget to show form for switched locale
+        $lockerData = $this->getLocaleSaveDataAsArray($locale) ?: [];
+
+        $this->reprocessLocaleItems($lockerData);
+
+        $this->formWidget->setFormValues($lockerData);
```

Comment:
Also here:
```
        foreach ($this->formWidgets as $key => $widget) {
            $value = array_shift($lockerData);
            if (!$value) {
                unset($this->formWidgets[$key]);
            }
            else {
                $widget->setFormValues($value);
            }
        }
```

Reply:
This code was taken from the MLRepeater and does not work for the nested form as it only has `$this->formWidget` available. Or I'm missing something.